ribbon-loadbalancer/src/main/java/com/netflix/loadbalancer/RoundRobinRule.java

```diff
@@ -17,52 +17,51 @@
  */
 package com.netflix.loadbalancer;
 
-import java.util.List;
-import java.util.concurrent.atomic.AtomicLong;
-
+import com.netflix.client.config.IClientConfig;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import com.netflix.client.config.IClientConfig;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
 
 /**
- * The most well known and basic loadbalacing strategy, i.e. Round Robin Rule.
- * 
+ * The most well known and basic load balancing strategy, i.e. Round Robin Rule.
+ *
  * @author stonse
- * 
+ * @author Nikos Michalakis <nikos@netflix.com>
+ *
  */
 public class RoundRobinRule extends AbstractLoadBalancerRule {
-    /** We use a long to never run into wraparound issues when this runs for a long time. */
-    private AtomicLong nextIndexAI;
+
+    /** Goes back to 0 when we reach Integer.MAX_VALUE. */
+    private AtomicInteger nextServerCyclicCounter;
+    /** Flag for selecting available servers. */
+    private final static boolean availableOnlyServers = true;
+    /** Flag for selecting every server. */
+    private final static boolean allServers = false;
 
     private static Logger log = LoggerFactory.getLogger(RoundRobinRule.class);
 
     public RoundRobinRule() {
-        nextIndexAI = new AtomicLong(0);
+        nextServerCyclicCounter = new AtomicInteger(0);
     }
 
     public RoundRobinRule(ILoadBalancer lb) {
-    	this();
-    	setLoadBalancer(lb);
+        this();
+        setLoadBalancer(lb);
     }
 
-    /*
-     * Rotate over all known servers.
-     */
-    final static boolean availableOnly = false;
-
     public Server choose(ILoadBalancer lb, Object key) {
         if (lb == null) {
             log.warn("no load balancer");
             return null;
         }
-        Server server = null;
-        int index = 0;
 
+        Server server = null;
         int count = 0;
         while (server == null && count++ < 10) {
-            List<Server> upList = lb.getServerList(true);
-            List<Server> allList = lb.getServerList(false);
+            List<Server> upList = lb.getServerList(availableOnlyServers);

```

Comment:
A 'cleaner' and more readable API would be to create two new methods in ILoadBalancer.  getUpServerList() and getServerList().  Then deprecated getServerList(boolean).